lua/bufferline/animate.lua

```diff
@@ -2,38 +2,18 @@
 -- animate.lua
 --
 
-local float2nr = vim.fn.float2nr
-local reltime = vim.fn.reltime
-local reltimefloat = vim.fn.reltimefloat
-
-
-
 local animation_frequency = 50
 
-function start(duration, initial, final, type, callback)
-  local ticks = (duration / animation_frequency) + 10
+local function lerp(ratio, initial, final, delta_type)
+  delta_type = delta_type or vim.v.t_number
 
-  local state = {}
-  state.running = true
-  state.fn = callback
-  state.type = type
-  state.step = (final - initial) / ticks
-  state.duration = duration
-  state.current = initial
-  state.initial = initial
-  state.final = final
-  state.start = reltime()
-  state.timer = vim.loop.new_timer()
-
-  state.timer:start(0, animation_frequency, vim.schedule_wrap(function()
-    animate_tick(state.timer, state)
-  end))
+  local range = final - initial
+  local delta = delta_type == vim.v.t_number and math.floor(ratio * range) or (ratio * range)
 
-  state.fn(state.current, state)
-  return state
+  return initial + delta
 end
 
-function animate_tick(timer, state)
+local function animate_tick(timer, state)
```

Comment:
You'll see me add `local` a lot on this PR. coc.nvim probably just didn't warn you about making functions global at the time, and I haven't noticed any repercussions for taking them out of the global namespace so far.

Reply:
Good change, I've been polluting the global namespace inadvertently.